src/test/java/org/folio/permstest/RestVerticleTest.java

```diff
@@ -1451,23 +1458,23 @@ public void handle(HttpClientResponse hcr) {
 
   //need test to find bad.delete and verify that it is a dummy perm
   private Future<WrappedResponse> testBadPermissionSet(TestContext context) {
-    Future<WrappedResponse> future = Future.future();
+    Promise<WrappedResponse> promise = Promise.promise();
     testPermissionExists(context, "bad.delete", true).onComplete( testRes -> {
       if(testRes.failed()) {
-        future.fail(testRes.cause());
+        promise.fail(testRes.cause());
       } else {
         WrappedResponse wr = testRes.result();
         JsonObject json = new JsonObject(wr.getBody());
         boolean dummy = json.getJsonArray("permissions").getJsonObject(0)
             .getBoolean("dummy");
         if(!dummy) {
-          future.fail("bad.delete is not flagged as a dummy perm");
+          promise.fail("bad.delete is not flagged as a dummy perm");
         } else {
-          future.complete(wr);
+          promise.complete(wr);
         }
       }
     });
-    return future;
+    return promise.future();
```

Comment:
```
    return testPermissionExists(context, "bad.delete", true)
      .compose(wr -> {
        JsonObject json = new JsonObject(wr.getBody());
        boolean dummy = json.getJsonArray("permissions").getJsonObject(0)
            .getBoolean("dummy");
        if(!dummy) {
          return Future.failedFuture("bad.delete is not flagged as a dummy perm");
        }
        return Future.succeededFuture(wr);
      });
```
https://github.com/folio-org/raml-module-builder/commits/master/doc/futurisation.md explains how to futurise when migrating to Vert.x 4.
Use `compose` to avoid an unneccessary Promise.

The same applies to many more methods below.

Reply:
Fixed.